app/controllers/hq/users_controller.rb

```diff
@@ -6,4 +6,24 @@ def index
     @users = User.all
   end
 
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    if @user.update_attributes(user_params)
+      flash[:success] = 'User successfully saved'
+      redirect_to hq_user_url(@user)
+    else
+      render 'edit'
+    end
+  end
+
+private
+
+  def user_params
+    params[:user].permit(:user_name, :email)
```

Comment:
Oh ok, sorry I thought you meant to require the mandatory attributes user_name and email.
So what, if anything, should I do about handling the exception?